src/mantine-hooks/src/use-focus-trap/scope-tab.ts

```diff
@@ -8,7 +8,19 @@ export function scopeTab(node: HTMLElement, event: KeyboardEvent) {
   }
   const finalTabbable = tabbable[event.shiftKey ? 0 : tabbable.length - 1];
   const root = node.getRootNode() as unknown as DocumentOrShadowRoot;
-  const leavingFinalTabbable = finalTabbable === root.activeElement || node === root.activeElement;
+  let leavingFinalTabbable = finalTabbable === root.activeElement || node === root.activeElement;
+
+  // Handle the case of the active element being in a RadioGroup with the finalTabbable element
+  if (root.activeElement instanceof HTMLInputElement && root.activeElement.type === 'radio') {
+    const activeRadioGroup = tabbable.filter(
+      (element) =>
+        element instanceof HTMLInputElement &&
+        root.activeElement instanceof HTMLInputElement &&
+        element.type === 'radio' &&
+        element.name === root.activeElement.name
+    );
+    leavingFinalTabbable = activeRadioGroup.includes(finalTabbable);
+  }
```

Comment:
This, obviously, is only a solution for Radio Groups, but I suspect there's similar behavior for other elements.

Another idea I had was to get the next target, and check if it's in `tabbable` at all, but I wasn't sure if there was an API exposed to do that easily, or if we'd need to do a lot of work to calculate it on our own.